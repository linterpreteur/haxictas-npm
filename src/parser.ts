export type MenuData = {
  data: {
      cafeteria: string,
      meals: {[item: string]: string|number}[]
  },
  date: Date
};

export type MenuParams = {data: string, date: Date};

export type CafeteriaData = {
  cafeteria: string,
  location: string,
  hours: {conditions: {day: string[], floor?: string, opens_at: string[], closes_at: string[]}}[]
};

export type CafeteriaParams = string;

export type Parser = {
  menus: (params: MenuParams) => Generator<MenuData, void, undefined>,
  cafeterias: (params: CafeteriaParams) => Generator<CafeteriaData, void, undefined>
}
