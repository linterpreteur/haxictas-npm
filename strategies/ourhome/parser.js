const cheerio = require('cheerio');
const prices = require('./prices');

module.exports.menus = ({data: page}, callback) => {
    const cached = this.cached = this.cached || {};
    const $ = cheerio.load(page);

    const parsePrice = (board) => {
        const price = {};
        const lis = board.find('li');
        if (lis.contents().length) return false;
        lis.each(li => {
            const tag = $(li).attr('class');
            const value = $(li).text();
            if (value.match(/[0-9]원/)) {
                value = parseInt(value.replace(/[^0-9]/, ''), 10);
            }
            price[tag] = value;
        });
        return price;
    };
    
    cached.price = parsePrice($('.board')) || cached.price || prices;
    
    const cafeterias = {};
    $('td[scope=rowgroup]').each((i, td) => {
        cafeterias[$(td).text().trim()] = i;
    });
    
    const dates = [];
    for (let i = 0; i < 7; i++) {
        const date = [];
        for (let j = 0; j < Object.keys(cafeterias).length; j++) {
            date.push({});
        }
        dates.push(date);
    }
    
    let day = 0; // 0 = SUN, 1 = MON, ..., 6 = SAT
    let meal = 0; // 0 = BFST, 1 = LNCH, 2 = DNNR
    let cafeteria = null;
    
    const meals = ['아침', '점심', '저녁'];
    let onRemarksCell;
            
    const isCafeteriaNameCell = (td) => $(td).attr('scope') === 'rowgroup';
    
    $('.t_col td').each((_, td) => {
        const lis = $(td).find('li');
        const cellText = $(td).text().trim();
        
        const cellEmpty = !cellText;
        const cellHasItems = !!lis.contents().length;
        const onTagCell = !cellEmpty && !cellHasItems;
        const onMenuItemCell = cellHasItems || (!onRemarksCell && cellEmpty);
        
        if (onTagCell) {
            onRemarksCell = (cellText === '비고');
            if (isCafeteriaNameCell(td)) cafeteria = cellText.trim();
            if (meals.indexOf(cellText) > -1) meal = meals.indexOf(cellText);
        } else if (onMenuItemCell) {
            const result = dates[day][cafeterias[cafeteria]];
            
            result['cafeteria'] = result['cafeteria'] || cafeteria;
            result['meals'] = result['meals'] || [{}, {}, {}];
            
            lis.each((_, li) => {
                const item = $(li).text().trim();
                const symbol = $(li).attr('class');
                const price = cached.price[symbol] || prices.none;
                
                result['meals'][meal][item] = price;
            });
            
            day = (day + 1) % 7;
        }
    });
    
    dates.forEach((date, day) => {
        date.forEach(cafeteria => {
            callback({day: day, data: cafeteria});
        });
    });
};

module.exports.cafeterias = (page, callback) => {
    const $ = cheerio.load(page);
    const tables = $('.basic_tbl').filter(i => i < 2);
    
    const onBreakPattern = /\(방학중 ([0-9]{2}:[0-9]{2})\)/;
    
    tables.each((i, table) => {
        const result = {
            cafeteria: i === 0 ? '919동' : '아워홈',
            hours: []
        };
        
        $(table).find('tr').each((i, tr) => {
            const tds = $(tr).find('td');
            if (i > 3) {
                result.location = tds.first().text().trim();
            } else {
                const search = function(days) {
                    for (let i = 0; i < result.hours.length; i++) {
                        const target = result.hours[i];
                        if (target.conditions.day.every(x => days.indexOf(x) != -1) &&
                            days.every(x => target.conditions.day.indexOf(x) != -1)) {
                            return target;
                        }
                    }
                    return null;
                };
                
                const update = function(days, hours) {
                    let found = search(days);
                    if (!found) {
                        found = {
                            conditions: {day: days},
                            opens_at: [],
                            closes_at: []
                        };
                        result.hours.push(found);
                    }
                    found.opens_at.push(hours[0].trim());
                    found.closes_at.push(hours[1].trim());
                };
                
                tds.each((i, td) => {
                    let text = $(td).text().trim();
                    const tokens = () => text.split('~');
                    let days = [i === 0 ? 'weekdays' : 'weekends'];
                    
                    if ($(td).attr('colspan') == 2) {
                        days = ['weekdays', 'weekends'];
                    }
                    
                    if (text.indexOf('방학중') != -1) {
                        const onBreakTokens = tokens().map(token => {
                            const match = token.match(onBreakPattern);
                            return (match && match[1]) || token;
                        });
                        
                        update(days.concat('break'), onBreakTokens);
                        
                        text = text.replace(onBreakPattern, '');
                    }
                    update(days, tokens());
                });
            }
        });
        callback(result);
    });
};
