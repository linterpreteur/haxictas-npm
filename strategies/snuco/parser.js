var cheerio = require('cheerio');
var prices = require('./prices');

module.exports.menus = (page, callback) => {
    var cached = this.cached = this.cached || {};
    var $ = cheerio.load(page);
    
    var pricePattern = /[\u24D0-\u24E9] [^ ]+/g;
    var parsePrice = (text) => {
        var price = {};
        var matches = text.match(pricePattern);
        if (!matches) return false;
        matches.forEach((m) => {
            var symbol = m.substring(0, m.indexOf(' '));
            var value = m.substring(m.indexOf(' ') + 1);
            if (value.match(/[0-9]원/)) {
                value = parseInt(value.replace(/[^0-9]/, ''), 10);
            }
            price[symbol] = value;
        });
        return price;
    };
    
    var trs = $('#Content tr');
    
    cached.price = parsePrice(trs.last().find('td').text()) || cached.price || prices;
    
    trs.filter((_, v) => $(v).text().trim()).each((i, tr) => {
        var tds = $(tr).find('td');
        
        var result = {
            cafeteria: null,
            meals: []
        };
        
        var priceSymbols = /[\u24D0-\u24E9]/g;
        var separators = /[/\n]+(?=^|[\u24D0-\u24E9])/;
        
        tds.filter(i => i % 2 === 0).each((i, td) => {
            td = $(td);
            var text = td.text();
            
            if (i === 0) {
                if (!text || text.match(pricePattern)) return false;
                td.html(td.html().replace(/(<.+?>.*)<br>.*(<.+?>)/, '$1$2'));
                result.cafeteria = td.text();
            } else {
                var items = {};
                var matches = text.match(priceSymbols);
                
                if (matches) {
                    text.replace(/\[.+?\]|\(.+?\)/, '')
                        .split(separators).forEach(text => {
                            var match = text.match(priceSymbols);
                            var symbol = match && match[0];
                            var item = text.replace(symbol, '').trim();
                            var price = cached.price[symbol] || prices.none;
                            items[item] = price;
                        });
                } else if (text.trim()) {
                    items[text.trim()] = prices.none;
                }
                result.meals.push(items);
            }
        });
        
        if (!result.cafeteria) return;
        
        callback(result);
    });
};

module.exports.cafeterias = (page, callback) => {
    var $ = cheerio.load(page);
    
    // hard coding hell yeah!
    // have no idea whats going on
    var cafeteria = 'cafeteria';
    var floor = 'floor';
    var days = ['weekdays', 'saturday', 'holidays'];
    var categories = [
        cafeteria,
        'location',
        floor,
        'chairs',
        'target',
        days[0],
        days[1],
        days[2]
    ];
    var secondLineColumnOffset = 2;
    
    var trs = $('#Content tr');
    
    trs.filter((_, v) => $(v).text().trim()).each((i, tr) => {
        var tds = $(tr).find('td');
        var mod = tds.first().text().trim() ? 0 : 1;
        var result = {
            hours: []
        };
        
        var _floor = floor;
        var inserted = {};
        tds.filter(i => i % 2 === mod).each((i, td) => {
            var category = categories[i + (mod === 0 ? 0 : secondLineColumnOffset)];
            td = $(td);
            if (category === cafeteria) {
                td.html(td.html().replace(/(\s+|\n)/g, ' '));
                td.html(td.html().replace(/(<.+?>.*)<br>.*(<.+?>)/, '$1$2'));
            }
            var text = td.text().trim();
            
            if (categories.indexOf(category) < secondLineColumnOffset) {
                result[category] = text;
            } else {
                if (category === floor) {
                    _floor = text;
                } if (days.indexOf(category) != -1) {
                    text = text.replace(/ ?[-~] ?/g, '-');
                    
                    var hour = function(regex) {
                        var matches = text.match(regex);
                        if (!matches) return null;
                        return matches.map(x => x.match(/[0-9:]+/)[0]);
                    };
                    
                    var opens_at = hour(/[0-9]{2}:[0-9]{2}\-/g);
                    var closes_at = hour(/\-[0-9]{2}:[0-9]{2}/g);
                    
                    var dupe = inserted[_floor + text];
                    if (isFinite(dupe)) {
                        result.hours[dupe].conditions.day.push(category);
                    } else if (opens_at || closes_at) {
                        result.hours.push({
                            conditions: {
                                floor: _floor,
                                day: [category]
                            },
                            opens_at: opens_at,
                            closes_at: closes_at
                        });
                        inserted[_floor + text] = result.hours.length - 1;
                    }
                }
            }
        });
        callback(result);
    });
};