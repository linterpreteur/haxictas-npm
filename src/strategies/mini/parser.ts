import {JSDOM} from 'jsdom';
import {MenuParams, MenuCallback} from '../../parser';
import {dedupe, querySelectorArray, normalize} from '../../lib/utils';

export function menus({data, date}: MenuParams, callback: MenuCallback) {
    const {document} = (new JSDOM(data)).window;

    const cafeterias = dedupe(
        querySelectorArray(document, 'td.head')
            .map((x) => normalize(x.textContent))
    );

    const menu: {[cafeteria: string]: {[item: string]: number}[]} = {};
    cafeterias.forEach(cafeteria => { menu[cafeteria] = [] });
    
    const cells = querySelectorArray(document, 'td');
    let meal = -1;
    let cafeteria = null;
    cells.forEach(cell => {
        const text = cell.textContent.trim();

        switch (text) {
            case '아침':
                meal = 0;
                return;
            case '점심':
                meal = 1;
                return;
            case '저녁':
                meal = 2;
                return;
        }

        const priceClass = 'price';
        const priceSelector = `.${priceClass}`;
        const isMenuItem = !!cell.querySelector(priceSelector);
        if (!isMenuItem) {
            cafeteria = text;
            return;
        }

        let price: number;
        cell.childNodes.forEach(child => {
            if (child instanceof child.ownerDocument.defaultView['Element']) {
                const isPrice = (child as Element).classList.contains(priceClass);
                if (!isPrice) {
                    return;
                }
                const priceSymbol = normalize(child.textContent);
                price = parseInt(priceSymbol, 10) * 100;
                return;
            }
            const item = normalize(child.textContent);
            menu[cafeteria][meal] = Object.assign(
                {},
                (menu[cafeteria][meal] || {}),
                { [item]: price }
            );
        });
    });

    Object.keys(menu).forEach(cafeteria => {
        const meals = menu[cafeteria]
        const data = {
            cafeteria: cafeteria,
            meals: meals
        }
        callback({
            data: data,
            date: date
        })
    })
};
