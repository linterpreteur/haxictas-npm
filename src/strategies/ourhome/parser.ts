import * as cheerio from 'cheerio';
import prices from './prices';
import {MenuParams, MenuCallback, CafeteriaParams, CafeteriaCallback, CafeteriaData} from '../../parser';

export function menus({data: page, date: startDate}: MenuParams, callback: MenuCallback) {
    const cached = this.cached = this.cached || {};
    const $ = cheerio.load(page);

    const parsePrice = (board: Cheerio) => {
        const price = {};
        const lis = board.find('li');
        if (lis.contents().length) return false;
        lis.each(li => {
            const tag = $(li).attr('class');
            let value = $(li).text();
            if (value.match(/[0-9]원/)) {
                value = value.replace(/[^0-9]/, '');
            }
            price[tag] = value;
        });
        return price;
    };
    
    cached.price = parsePrice($('.board')) || cached.price || prices;
    
    const cafeterias = {};
    $('td[scope=rowgroup]').each((i, td) => {
        cafeterias[$(td).text().trim()] = i;
    });
    
    const dates: {cafeteria: string, meals: {[item: string]: number}[]}[][] = [];
    for (let i = 0; i < 7; i++) {
        const date = [];
        for (let j = 0; j < Object.keys(cafeterias).length; j++) {
            date.push({});
        }
        dates.push(date);
    }
    
    let day = 0; // 0 = SUN, 1 = MON, ..., 6 = SAT
    let meal = 0; // 0 = BFST, 1 = LNCH, 2 = DNNR
    let cafeteria = null;
    
    const meals = ['아침', '점심', '저녁'];
    let onRemarksCell: boolean;
            
    const isCafeteriaNameCell = (td) => $(td).attr('scope') === 'rowgroup';
    
    $('.t_col td').each((_, td) => {
        const lis = $(td).find('li');
        const cellText = $(td).text().trim();
        
        const cellEmpty = !cellText;
        const cellHasItems = !!lis.contents().length;
        const onTagCell = !cellEmpty && !cellHasItems;
        const onMenuItemCell = cellHasItems || (!onRemarksCell && cellEmpty);
        
        if (onTagCell) {
            onRemarksCell = (cellText === '비고');
            if (isCafeteriaNameCell(td)) cafeteria = cellText.trim();
            if (meals.indexOf(cellText) > -1) meal = meals.indexOf(cellText);
        } else if (onMenuItemCell) {
            const result = dates[day][cafeterias[cafeteria]];
            
            result['cafeteria'] = result['cafeteria'] || cafeteria;
            result['meals'] = result['meals'] || [{}, {}, {}];
            
            lis.each((_, li) => {
                const item = $(li).text().trim();
                const symbol = $(li).attr('class');
                const price = cached.price[symbol] || prices.none;
                
                result['meals'][meal][item] = price;
            });
            
            day = (day + 1) % 7;
        }
    });
    
    dates.forEach((date, day) => {
        date.forEach(cafeteria => {
            const menuDate = new Date(startDate);
            menuDate.setDate(startDate.getDate() + (day - startDate.getDay()) % 7);
            callback({date: menuDate, data: cafeteria});
        });
    });
};

export function cafeterias(page: CafeteriaParams, callback: CafeteriaCallback) {
    const $ = cheerio.load(page);
    const tables = $('.basic_tbl').filter(i => i < 2);
    
    const onBreakPattern = /\(방학중 ([0-9]{2}:[0-9]{2})\)/;
    
    tables.each((i, table) => {
        const result: CafeteriaData = {
            cafeteria: i === 0 ? '919동' : '아워홈',
            hours: [],
            location: undefined
        };
        
        $(table).find('tr').each((i, tr) => {
            const tds = $(tr).find('td');
            if (i > 3) {
                result.location = tds.first().text().trim();
            } else {
                const search = function(days: string[]) {
                    for (let i = 0; i < result.hours.length; i++) {
                        const target = result.hours[i];
                        if (target.conditions.day.every(x => days.indexOf(x) != -1) &&
                            days.every(x => target.conditions.day.indexOf(x) != -1)) {
                            return target;
                        }
                    }
                    return null;
                };
                
                const update = function(days: string[], hours: string[]) {
                    let found = search(days);
                    if (!found) {
                        found = {
                            conditions: {
                                day: days,
                                opens_at: [],
                                closes_at: []
                            },
                        };
                        result.hours.push(found);
                    }
                    found.conditions.opens_at.push(hours[0].trim());
                    found.conditions.closes_at.push(hours[1].trim());
                };
                
                tds.each((i, td) => {
                    let text = $(td).text().trim();
                    const tokens = () => text.split('~');
                    let days = [i === 0 ? 'weekdays' : 'weekends'];
                    
                    if (parseInt($(td).attr('colspan'), 10) === 2) {
                        days = ['weekdays', 'weekends'];
                    }
                    
                    if (text.indexOf('방학중') != -1) {
                        const onBreakTokens = tokens().map(token => {
                            const match = token.match(onBreakPattern);
                            return (match && match[1]) || token;
                        });
                        
                        update(days.concat('break'), onBreakTokens);
                        
                        text = text.replace(onBreakPattern, '');
                    }
                    update(days, tokens());
                });
            }
        });
        callback(result);
    });
};
