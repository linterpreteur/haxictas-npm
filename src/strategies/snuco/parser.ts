import {JSDOM} from 'jsdom';
import {normalize, querySelectorArray} from '../../lib/utils';
import {MenuParams, MenuCallback, CafeteriaParams, CafeteriaCallback, CafeteriaData} from '../../parser';

export function menus({data: page, date}: MenuParams, callback: MenuCallback) {
    const {document} = (new JSDOM(page)).window;

    function parseMenu(s: string): {[item: string]: number} {
        return s.split(/\n+/g)
            .map(x => x.match(/(.+?)\s*([\d,.]+원)$/) || x)
            .map(x => (typeof x === 'string') ? [x, '기타'] : [x[1], x[2]])
            .map(([k, v]) => [normalize(k), normalize(v)])
            .filter(x => x && x[0])
            .reduce((acc, [k, v]) => Object.assign({}, acc, {[k]: v}), {});
    }
    querySelectorArray(document, 'tbody tr').map(row => {
        const [cafeteria, breakfast, lunch, dinner] = querySelectorArray(row, 'td');
        const data = {
            cafeteria: normalize(cafeteria.textContent),
            meals: [
                parseMenu(breakfast.textContent),
                parseMenu(lunch.textContent),
                parseMenu(dinner.textContent)
            ]
        };
        callback({data: data, date: date});
    });
};

export function cafeterias(page: CafeteriaParams, callback: CafeteriaCallback) {
    const {document} = (new JSDOM(page)).window;

    const header = querySelectorArray(document, 'thead th:not([colspan])');
    const nameIndex = 0;
    const locationIndex = 1;

    const info: CafeteriaData = {
        cafeteria: undefined,
        location: undefined,
        hours: undefined
    };
    let cafeteria = null;
    let floor = null;
    querySelectorArray(document, 'tbody tr').forEach(row => {
        const cells = querySelectorArray(row, 'td');

        const isFullRow = cells.length >= header.length;
        if (isFullRow) {
            const name = cells[nameIndex];
            cafeteria = normalize(name.textContent);

            const location = cells[locationIndex];
            info[cafeteria] = Object.assign(
                {},
                info[cafeteria] || {},
                { location: normalize(location.textContent) }
            );

            cells.splice(cells.findIndex(x => x === name), 1);
            cells.splice(cells.findIndex(x => x === location), 1);
        }

        const mayContainFloor = cells.length > 5;
        if (mayContainFloor) {
            floor = normalize(cells[0].textContent);
            cells.splice(0, 1);
        }
        
        const [size, _, weekdays, saturday, holidays] = cells;

        const specialTypes = /(채식)/;
        const specialTypeMatches = size.textContent.match(specialTypes);
        if (specialTypeMatches) {
            floor = normalize(specialTypeMatches[1]);
        }

        function parseHours(s) {
            const numeralHours = /(\d+:\d+)-(\d+:\d+)/;
            const specialHours = /분식/;
            if (!s.match(numeralHours)) {
                return [normalize(s)];
            }
            if (s.match(specialHours)) {
                return [normalize(s)];
            }
            return s.split(/\n+/)
                .map(normalize)
                .map(x => x.match(numeralHours))
                .filter(x => x)
                .map(x => [x[1], x[2]]);
        }
        const hours = {
            weekdays: parseHours(weekdays.textContent),
            saturday: parseHours(saturday.textContent),
            holidays: parseHours(holidays.textContent)
        };

        function isEqualEntry(a, b) {
            return a.floor === b.floor &&
                a['opens_at'] === b['opens_at'] &&
                a['closes_at'] === b['closes_at'];
        }
        const aggregatedHours = Object.keys(hours)
            .reduce((acc, day) => {
                const newEntry = hours[day].map(item => ({
                    day: [day],
                    floor: floor,
                    'opens_at': (typeof item === 'string') ? item : item[0],
                    'closes_at': (typeof item === 'string') ? item : item[1],
                }));
                return acc.concat(newEntry);
            }, [])
            .reduce((acc, x) => {
                const equalEntry = acc.find(y => isEqualEntry(x, y));
                if (!equalEntry) {
                    return acc.concat(x);
                }
                equalEntry.day.push(...x.day);
                return acc;
            }, [])
            .map(x => ({conditions: x}));
        info[cafeteria] = Object.assign(
            {},
            info[cafeteria] || {},
            { hours: aggregatedHours }
        );
    });
    callback(info);
};
