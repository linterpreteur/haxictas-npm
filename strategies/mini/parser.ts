import {JSDOM} from 'jsdom';
import {MenuParams, MenuCallback} from '../../parser';
import {dedupe, querySelectorArray} from '../../lib/utils';

export function menus({data, date}: MenuParams, callback: MenuCallback) {
    const {document} = (new JSDOM(data)).window;

    const cafeterias = dedupe(
        querySelectorArray(document, 'td.head')
            .map((x) => x.textContent.replace(/\n/g, '').trim())
    );

    const menu: {[cafeteria: string]: {[item: string]: number}[]} = {};
    cafeterias.forEach(cafeteria => { menu[cafeteria] = [] });
    
    const cells = querySelectorArray(document, 'td');
    let meal = -1;
    let cafeteria = null;
    cells.forEach(cell => {
        const text = cell.textContent.trim();

        switch (text) {
            case '아침':
                meal = 0;
                return;
            case '점심':
                meal = 1;
                return;
            case '저녁':
                meal = 2;
                return;
        }

        const isMenuItem = text.match(/^\d+ /)
        if (!isMenuItem) {
            cafeteria = text;
            return;
        }

        const priceSelector = '.price';
        const priceSymbol = cell.querySelector(priceSelector).textContent.trim();
        const price = parseInt(priceSymbol, 10) * 100;
        const item = (() => {
            const tmp = cell.cloneNode(true) as typeof cell;
            const priceNode = tmp.querySelector(priceSelector);
            tmp.removeChild(priceNode);
            return tmp.textContent.replace(/\s{2,}/g, ' ').trim();
        })();
        menu[cafeteria][meal] = Object.assign(
            {},
            (menu[cafeteria][meal] || {}),
            { [item]: price }
        )
    });

    Object.keys(menu).forEach(cafeteria => {
        const meals = menu[cafeteria]
        const data = {
            cafeteria: cafeteria,
            meals: meals
        }
        callback({
            data: data,
            date: date
        })
    })
};

export function cafeterias() { /* no-op */ };
