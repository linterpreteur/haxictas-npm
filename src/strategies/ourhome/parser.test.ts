import {ourhome} from '../../test/data';
import {menus, cafeterias} from './parser';

describe(menus, () => {
  it('parses given page', () => {
    const params = { data: ourhome.menus, date: new Date(0) };

    menus(params, (x, err) => {

      expect(x).toMatchSnapshot();
      expect(err).not.toBe(expect.anything());
    });
  });
});

describe(cafeterias, () => {
  it('parses given page', () => {
    const params = ourhome.cafeterias;

    cafeterias(params, (x, err) => {

      expect(x).toMatchSnapshot();
      expect(err).not.toBe(expect.anything());
    });
  });
});
