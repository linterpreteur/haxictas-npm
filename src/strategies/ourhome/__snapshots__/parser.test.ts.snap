// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cafeterias parses given page 1`] = `
Object {
  "cafeteria": "919동",
  "hours": Array [
    Object {
      "conditions": Object {
        "closes_at": Array [
          "09:30",
        ],
        "day": Array [
          "weekdays",
          "break",
        ],
        "opens_at": Array [
          "08:00",
        ],
      },
    },
    Object {
      "conditions": Object {
        "closes_at": Array [
          "09:30",
          "13:30",
          "19:30",
        ],
        "day": Array [
          "weekdays",
        ],
        "opens_at": Array [
          "07:30",
          "11:30",
          "17:30",
        ],
      },
    },
    Object {
      "conditions": Object {
        "closes_at": Array [
          "09:30",
          "13:30",
          "19:30",
        ],
        "day": Array [
          "weekends",
        ],
        "opens_at": Array [
          "08:00",
          "11:30",
          "17:30",
        ],
      },
    },
  ],
  "location": "919A동 지하 1층",
}
`;

exports[`cafeterias parses given page 2`] = `
Object {
  "cafeteria": "아워홈",
  "hours": Array [
    Object {
      "conditions": Object {
        "closes_at": Array [
          "09:30",
        ],
        "day": Array [
          "weekdays",
          "break",
        ],
        "opens_at": Array [
          "08:00",
        ],
      },
    },
    Object {
      "conditions": Object {
        "closes_at": Array [
          "09:30",
        ],
        "day": Array [
          "weekdays",
        ],
        "opens_at": Array [
          "07:30",
        ],
      },
    },
    Object {
      "conditions": Object {
        "closes_at": Array [
          "09:30",
        ],
        "day": Array [
          "weekends",
        ],
        "opens_at": Array [
          "08:00",
        ],
      },
    },
    Object {
      "conditions": Object {
        "closes_at": Array [
          "13:30",
          "19:30",
        ],
        "day": Array [
          "weekdays",
          "weekends",
        ],
        "opens_at": Array [
          "11:30",
          "17:30",
        ],
      },
    },
  ],
  "location": "901동 1층",
}
`;

exports[`menus parses given page 1`] = `
Object {
  "data": Object {
    "cafeteria": "아워홈",
    "meals": Array [
      Object {
        "사골우거지국": 3000,
        "토스트,씨리얼": 2500,
      },
      Object {
        "얼큰감자수제비&햄전": 4000,
        "제육장조림버터비빔밥&해시브라운": 4000,
      },
      Object {
        "뚝배기제육불고기": 4000,
      },
    ],
  },
  "date": 1969-12-28T00:00:00.000Z,
}
`;

exports[`menus parses given page 2`] = `
Object {
  "data": Object {
    "cafeteria": "919동",
    "meals": Array [
      Object {
        "쇠고기브로콜리볶음": 3000,
      },
      Object {
        "설렁탕": 3000,
        "핫커리치킨까스": 4000,
      },
      Object {
        "두부와쇠고기볶음": 3000,
      },
    ],
  },
  "date": 1969-12-28T00:00:00.000Z,
}
`;

exports[`menus parses given page 3`] = `
Object {
  "data": Object {
    "cafeteria": "아워홈",
    "meals": Array [
      Object {
        "콩비지찌개": 3000,
        "토스트씨리얼우유": 2500,
      },
      Object {
        "제주돼지김치짜글이&라면사리": 4000,
        "통베이컨볶음밥&계란후라이": 4000,
      },
      Object {
        "돌솥치즈닭갈비덮밥": 4000,
      },
    ],
  },
  "date": 1969-12-29T00:00:00.000Z,
}
`;

exports[`menus parses given page 4`] = `
Object {
  "data": Object {
    "cafeteria": "919동",
    "meals": Array [
      Object {
        "무쇠고기국&생선까스": 3000,
      },
      Object {
        "제육쭈꾸미볶음": 4000,
        "콩나물국밥": 2500,
      },
      Object {
        "모듬버섯불고기": 3500,
        "새우튀김돈까스벤또": 4000,
      },
    ],
  },
  "date": 1969-12-29T00:00:00.000Z,
}
`;

exports[`menus parses given page 5`] = `
Object {
  "data": Object {
    "cafeteria": "아워홈",
    "meals": Array [
      Object {
        "순두부찌개": 3000,
        "토스트씨리얼우유": 2500,
      },
      Object {
        "잔치국수&김밥": 4000,
        "카레돈까스": 4000,
      },
      Object {
        "에그함박스테이크": 4000,
      },
    ],
  },
  "date": 1969-12-30T00:00:00.000Z,
}
`;

exports[`menus parses given page 6`] = `
Object {
  "data": Object {
    "cafeteria": "919동",
    "meals": Array [
      Object {
        "단호박밥&불고기": 3000,
      },
      Object {
        "고등어김치조림": 3500,
        "돌솥갈비탕": 3000,
      },
      Object {
        "(5500원)아보카도명란마요덮밥&육즙함박": "기타",
        "순살양념치킨": 3500,
      },
    ],
  },
  "date": 1969-12-30T00:00:00.000Z,
}
`;

exports[`menus parses given page 7`] = `
Object {
  "data": Object {
    "cafeteria": "아워홈",
    "meals": Array [
      Object {
        "나물비빔밥": 3000,
        "토스트씨리얼우유": 2500,
      },
      Object {
        "★HAPPY NEW YEAR★": "기타",
        "새해에는떡만두국&모듬전": 4000,
      },
      Object {
        "참치김치찌개": 4000,
      },
    ],
  },
  "date": 1969-12-31T00:00:00.000Z,
}
`;

exports[`menus parses given page 8`] = `
Object {
  "data": Object {
    "cafeteria": "919동",
    "meals": Array [
      Object {
        "두부명란찌개": 2500,
      },
      Object {
        "갈비구이": 4000,
        "사골떡만두국": 3000,
      },
      Object {
        "해물잡채덮밥": 3000,
      },
    ],
  },
  "date": 1969-12-31T00:00:00.000Z,
}
`;

exports[`menus parses given page 9`] = `
Object {
  "data": Object {
    "cafeteria": "아워홈",
    "meals": Array [
      Object {},
      Object {},
      Object {},
    ],
  },
  "date": 1970-01-01T00:00:00.000Z,
}
`;

exports[`menus parses given page 10`] = `
Object {
  "data": Object {
    "cafeteria": "919동",
    "meals": Array [
      Object {
        "어묵김치국": 3000,
        "제육김치두루치기": 3000,
        "토스트씨리얼우유": 2500,
      },
      Object {
        "나가사키짬뽕&미니날치알밥": 5000,
        "닭개장&당면사리": 4000,
        "돌솥순두부계란동동": 3000,
        "치즈돈까스&쫄면": 4000,
      },
      Object {
        "밀푀유나베": 4000,
        "지코바치킨덮밥&누룽지샐러드": 4000,
        "훈제오리볶음밥": 3500,
      },
    ],
  },
  "date": 1970-01-01T00:00:00.000Z,
}
`;

exports[`menus parses given page 11`] = `
Object {
  "data": Object {
    "cafeteria": "아워홈",
    "meals": Array [
      Object {
        "얼큰수제비국&해물완자고추장조림": 3000,
        "토스트씨리얼우유": 2500,
      },
      Object {
        "뚝배기버섯불고기&당면사리": 4000,
        "치즈오븐스파게티": 4000,
      },
      Object {
        "돌솥제육비빔밥": 4000,
      },
    ],
  },
  "date": 1970-01-02T00:00:00.000Z,
}
`;

exports[`menus parses given page 12`] = `
Object {
  "data": Object {
    "cafeteria": "919동",
    "meals": Array [
      Object {
        "어묵탕": 2500,
      },
      Object {
        "돌솥감자탕": 3000,
        "불닭마요덮밥": 3500,
      },
      Object {
        "(5500원)레몬크림새우&짜장면": "기타",
        "오삼불고기": 3000,
      },
    ],
  },
  "date": 1970-01-02T00:00:00.000Z,
}
`;

exports[`menus parses given page 13`] = `
Object {
  "data": Object {
    "cafeteria": "아워홈",
    "meals": Array [
      Object {
        "콩나물해장국&계란후라이": 3000,
        "토스트씨리얼우유": 2500,
      },
      Object {
        "닭곰탕&소면사리/미트볼케찹조림": 4000,
        "마파두부덮밥": 4000,
      },
      Object {
        "제육두부찌개&생선까스": 4000,
      },
    ],
  },
  "date": 1970-01-03T00:00:00.000Z,
}
`;

exports[`menus parses given page 14`] = `
Object {
  "data": Object {
    "cafeteria": "919동",
    "meals": Array [
      Object {
        "해물된장찌개": 3000,
      },
      Object {
        "쇠고기버섯전골": 3000,
        "안동식짐닭&당면사리": 4000,
      },
      Object {
        "마라우육덮밥": 3500,
      },
    ],
  },
  "date": 1970-01-03T00:00:00.000Z,
}
`;
